internal/testutils/storeutils/store.go

```diff
@@ -0,0 +1,28 @@
+package storeutils
+
+import (
+	"os"
+	"testing"
+
+	"github.com/docker/docker/libnetwork/datastore"
+	"gotest.tools/v3/assert"
+)
+
+// NewTempStore creates a new temporary libnetwork store for testing purposes.
+// The store is created in a temporary directory that is cleaned up when the
+// test finishes.
+func NewTempStore(t *testing.T) *datastore.Store {
+	t.Helper()
+
+	tempDir, err := os.MkdirTemp("", "*")
+	assert.NilError(t, err)
+
+	ds, err := datastore.New(tempDir, "libnetwork")
+	assert.NilError(t, err)
+
+	t.Cleanup(func() {
+		assert.NilError(t, os.RemoveAll(tempDir))
+	})
```

Comment:
```suggestion
	ds, err := datastore.New(t.TempDir(), "libnetwork")
	assert.NilError(t, err)
```

Reply:
> TempDir returns the default directory to use for temporary files.

If we do that, all tests are going to use the same boltdb file. This could result in weird test failures, and this would prevent parallel execution.

I think many tests can't be parallelized currently because they share the same netns but let's not make it worth 🙂

Reply:
> [Each subsequent call to t.TempDir returns a unique directory](https://pkg.go.dev/testing#T.TempDir)

Reply:
Forget what I said, I thought you were suggesting `os.TempDir()` 🙈 I'll do that change!